### PR TITLE
Slightly better backoff for merge selecting task

### PR DESCRIPTION
**Pull request by tavplubix**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improved backoff for background cleanup tasks in `MergeTree`. Settings `merge_tree_clear_old_temporary_directories_interval_seconds` and `merge_tree_clear_old_parts_interval_seconds` moved form users settings to merge tree settings.

### CODE REVIEW

File: src/Core/Settings.h [hunk 1]
```diff
@@ -149,8 +149,6 @@ class IColumn;
     M(UInt64, merge_tree_coarse_index_granularity, 8, "If the index segment can contain the required keys, divide it into as many parts and recursively check them.", 0) \
     M(UInt64, merge_tree_max_rows_to_use_cache, (128 * 8192), "The maximum number of rows per request, to use the cache of uncompressed data. If the request is large, the cache is not used. (For large queries not to flush out the cache.)", 0) \
     M(UInt64, merge_tree_max_bytes_to_use_cache, (192 * 10 * 1024 * 1024), "The maximum number of bytes per request, to use the cache of uncompressed data. If the request is large, the cache is not used. (For large queries not to flush out the cache.)", 0) \
-    M(UInt64, merge_tree_clear_old_temporary_directories_interval_seconds, 60, "The period of executing the clear old temporary directories operation in background.", 0) \
```
Comment (nikitamikhaylov) [inline src/Core/Settings.h:152]: Better to mark them as deprecated and maybe log some warning if they are changed

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
I am really interested to see test result, because we have some aggressive ones such as `test_replicated_mutations/test.py::test_mutations_dont_prevent_merges`

File: src/Core/Settings.h (hunk 1 above)
Comment (tavplubix) [reply src/Core/Settings.h:152]: #31252

**Comment by tavplubix** [comment]
Stress tests (debug, actions) - #30692
Integration tests are broken in master.
`test_replicated_mutations/test.py::test_mutations_dont_prevent_merges` passed

**Pull request merged**